lib/chat.js

```diff
@@ -1,55 +1,124 @@
 var Chat = {
 
+	send_global: function(player, text){
+		Logger.debug('Chat.send_global');
+
+		if (!player.initials){
+			Logger.warning('Sending message without from_initials');
+			return;
+		}
+
+		if (!text){
+			Logger.warning('Sending message without text');
+			return;
+
+		}
+		
+		var packet = {
+			from: {
+				initials: player.initials
+			},
+			text: text
+		};
+		if (player.team){
+			packet.from.team = player.team;
+		}
+
+		player.socket.broadcast.emit('message.global', packet);
+	},
+
+	send_team: function(player, to_team, text){
+		Logger.debug('Chat.send_team');
+		if (!player.initials){
+			Logger.warning('Sending a team message without initials');
+			return;
+		}
+
+		if (!player.team){
+			Logger.warning('Sending a team message without a team');
+			return;
+		}
+
+		if (!to_team){
+			Logger.warning('Sending message without to_team');
+			return;
+		}
+
+		if (!text){
+			Logger.warning('Sending message without text');
+			return;
+		}
+
+		var packet = {
+			from: {
+				initials: player.initials
+			},
+			text: text
+		};
+		if (player.team){
+			packet.from.team = player.team;
+		}
+
+		player.socket.broadcast.to(to_team).emit('message.team', packet);
+	},
+
+	send_from_system: function(text){
+		Logger.debug('Chat.send_from_system');
+		if (!text){
+			Logger.warning('Sending message without text');
+		}
+
+		var packet = {
+			text: text
+		};
+
+		Socket.io.sockets.emit('message.system', packet);
+	},
+
 	// takes in a socket.io socket object. Enables chat protocol on this socket: message.global, message.team
 	enable: function(socket){
-		if (!socket || !socket.on){
+		Logger.debug('Chat.enable');
+		if (!socket){
+			Logger.warning('Chat.enable without a socket');
 			return;
 		}
 
+		// Event: incoming message to global. Might be directed at another team.
+		// Certainly not a team message (to own team)
 		socket.on('message.global', function(data){
-			if (!data || !data.text || !socket.initials){ // can't send nothing or send without initials.
+			Logger.debug('Event: message.global');
+
+			if (!data || !data.text || !socket.player || !socket.player.initials){ // can't send nothing or send without initials.
 				return;
 			}
 
-			if ((Timer.now() - socket.last_message) < 10){
+			var player = socket.player;
+
+			if (!player.last_message){
```

Comment:
make sure to do flood detection for team messages as well. Also return an event if the user is flooded, so the UI can show this.